plugins/org.eclipse.emf.common/src/org/eclipse/emf/common/util/Pool.java

```diff
@@ -623,11 +727,11 @@ else if (expectedEntry.get() == null && !TEST_ENQUEUED.isEnqueued(expectedEntry)
   }
 
   @Override
-  protected boolean ensureCapacity()
+  protected synchronized boolean ensureCapacity()
```

Comment:
@merks: One problem of this implementation as it stands in this PR is that it can deadlock when ensureCapacity callls grow. I plan to solve that in the next iteration of the PR.

I wanted to publish the PR to let you know what I have in mind.